views/js/pciCreator/ims/mathEntryInteraction/runtime/mathEntryInteraction.js

```diff
@@ -0,0 +1,835 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016-2021 (original work) Open Assessment Technologies SA;
+ *
+ * @author Christophe Noël <christophe@taotesting.com>
+ *
+ */
+define('mathEntryInteractionFactory', [
```

Comment:
It is more clear for me, because `qtiCustomInteractionContext` part is the IMS PCI API, that is public. The factory part is private and it can be called only through the API. And this is how I visually separated. If you do not like it I can move it together, just let me know.

Reply:
It would be better indeed to have all in the same module. It will be less confusing.

Reply:
done